src/random.rs

```diff
@@ -0,0 +1,722 @@
+    stream: StreamOrDevice,
+) -> Result<Array, Exception> {
+    let shape = shape.into().unwrap_or(&[]);
+
+    let key = key.into().map_or_else(
+        || {
+            let mut state = state().lock().unwrap();
+            state.next()
+        },
+        |key| key.clone(),
+    );
+
+    unsafe {
+        let c_array = try_catch_c_ptr_expr! {
+            mlx_sys::mlx_random_normal(
+                shape.as_ptr(),
+                shape.len(),
+                T::DTYPE.into(),
+                loc.into().unwrap_or(0.0),
+                scale.into().unwrap_or(1.0),
+                key.as_ptr(),
+                stream.as_ptr(),
+            )
+        };
+        Ok(Array::from_ptr(c_array))
+    }
+}
+
+/// Generate jointly-normal random samples given a mean and covariance.
+///
+/// The matrix `covariance` must be positive semi-definite. The behavior is
+/// undefined if it is not.  The only supported output type is f32.
+///
+/// # Params
+/// - mean: array of shape `[..., n]`, the mean of the distribution.
+/// - covariance: array  of shape `[..., n, n]`, the covariance
+/// matrix of the distribution. The batch shape `...` must be
+/// broadcast-compatible with that of `mean`.
+/// - shape: The output shape must be
+/// broadcast-compatible with `mean.shape.dropLast()` and `covariance.shape.dropLast(2)`.
+/// If empty, the result shape is determined by broadcasting the batch
+/// shapes of `mean` and `covariance`.
+/// - key: PRNG key
+#[default_device]
+pub fn multivariate_normal_device<'a, T: ArrayElement>(
+    mean: &Array,
+    covariance: &Array,
+    shape: impl Into<Option<&'a [i32]>>,
+    key: impl Into<Option<&'a Array>>,
+    stream: StreamOrDevice,
+) -> Result<Array, Exception> {
+    let shape = shape.into().unwrap_or(&[]);
+
+    let key = key.into().map_or_else(
+        || {
+            let mut state = state().lock().unwrap();
+            state.next()
+        },
+        |key| key.clone(),
+    );
+
+    unsafe {
+        let c_array = try_catch_c_ptr_expr! {
+            mlx_sys::mlx_random_multivariate_normal(
+                mean.as_ptr(),
+                covariance.as_ptr(),
+                shape.as_ptr(),
+                shape.len(),
+                T::DTYPE.into(),
+                key.as_ptr(),
+                stream.as_ptr(),
+            )
+        };
+        Ok(Array::from_ptr(c_array))
+    }
+}
+
+/// Generate random integers from the given interval (`lower_bound:` and `upper_bound:`).
+///
+/// The values are sampled with equal probability from the integers in
+/// half-open interval `[lb, ub)`. The lower and upper bound can be
+/// scalars or arrays and must be roadcastable to `shape`.
+///
+/// ```rust
+/// use mlx_rs::{array, random};
+///
+/// let key = random::key(0);
+///
+/// // generate an array of Int values, one in the range 0 ..< 10
+/// // and one in the range 10 ..< 100
+/// let array = random::randint::<_, i32>(array![0, 10], array![10, 100], None, &key);
```

Comment:
Is there a better example? Maybe one in the range [0, 20), and another in the range [10, 100)? I think the current one seems confusing like someone may think the first argument specifies both lb and ub for the first rand int